src/operator/mxnet_op.h

```diff
@@ -345,6 +394,13 @@ __global__ void mxnet_generic_kernel(int N, Args... args) {
   }
 }
 
+template<typename OP, typename ...Args>
+__global__ void mxnet_generic_kernel_ex(int N, Args... args) {
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += blockDim.x * gridDim.x) {
+    OP::Map(i, 1, args...);
```

Comment:
For GPU, can we also get the number of threads first, and set the corresponding length here as you did for CPU? I think it will also accelerate computation on GPU side because of less calls to the functions `unravel` and `dot`, right?

Reply:
GPU memory access works differently. memory access is fastest when adjacent threads access adjacent memory.

Reply:
In the case slice op, I benchmarked two different approaches for GPU: one is each thread working on a single element (frequent unravel and ravel calls like here with length = 1), the other one is similar to what you did for CPU, it turns out that the latter is a little bit faster. There are about 50,000 slice op's output elements.

Reply:
For GPU the fastest way is to add a step. Each thread work on N elements separated by step